examples/matrix/src/matrix.c

```diff
@@ -0,0 +1,124 @@
+// Copyright 2017 The Bazel Authors. All rights reserved.
```

Comment:
Yes, this is original code. I wanted a small example that tested an interesting set of FFI features, such as wrapping a C struct rather than just an opaque pointer, making a safe Rust wrapper over the C API, etc. I'm planning to reuse this example elsewhere, such as for rules_d, etc.